python-squid/src/test/java/org/sonar/python/DocstringExtractorTest.java

```diff
@@ -74,19 +74,22 @@ private Token getDocstring(PythonGrammar nodeType, int line) {
 
     @Override
     public void scanFile(PythonVisitorContext visitorContext) {
-      scanNode(visitorContext.rootAstNode());
-
+      scanTree(visitorContext.rootTree());
```

Comment:
I don't think we need a separate method here. This scanTree method can be inlined I believe.

Reply:
Tbh I'd like to keep it that way because scanTree recursively calls itself. If I inline, the benefit of getting rid of the method would be offset by the fact I'd need to reimplement the recursion through a loop/stack construct.

Reply:
missed the recursion, fair enough. But then I wonder why not make this a PythonCheckTree to avoid to reimplement the visit ?

Reply:
I reproduced the previous implementation based on AstNode without questioning it... Sure it makes more sense to do it that way now.